src/org/etools/j1939_84/controllers/part01/Part01Step14Controller.java

```diff
@@ -88,59 +87,80 @@ protected void run() throws Throwable {
             getListener().onResult("");
         }
 
+        // 6.1.14.2.a. Fail if any response for any monitor supported in DM5 by a given ECU is reported as
+        // “0=monitor complete this cycle or not supported” in SP 3303 bits 1-4 and SP 3305
+        // [except comprehensive components monitor (CCM)].
+        globalPackets.stream()
+                     .flatMap(p -> p.getMonitoredSystems().stream())
+                     .sorted()
+                     .filter(dm26System -> dm26System.getId() != CompositeSystem.COMPREHENSIVE_COMPONENT)
+                     .filter(dm26System -> {
+                         MonitoredSystem dm5System = getDM5System(dm26System.getId(), dm26System.getSourceAddress());
+                         return dm5System != null
+                                 && dm5System.getStatus().isEnabled()
+                                 && dm26System.getStatus().isComplete();
+
+                     })
+                     .forEach(dm26System -> {
+                         String moduleName = Lookup.getAddressName(dm26System.getSourceAddress());
+                         String systemName = dm26System.getName().trim();
+                         addFailure("6.1.14.2.a - " + moduleName + " response for a monitor " + systemName
+                                 + " in DM5 is reported as supported and is reported as complete/not supported DM26 response");
+                     });
+
+        // 6.1.14.2.b. Fail if any response for each monitor not supported in DM5 by a given ECU is not also reported
+        // in DM26 as “0=monitor complete this cycle or not supported” in SP 3303 bits 5-7 and
+        // [also reported in DM26 as] “0=monitor disabled for rest of this cycle or not supported” in
+        // SP 3303 bits 1 and 2 and SP 3304.
         globalPackets.stream()
                      .flatMap(p -> p.getMonitoredSystems().stream())
                      .sorted()
+                     .filter(dm26System -> {
+                         MonitoredSystem dm5System = getDM5System(dm26System.getId(), dm26System.getSourceAddress());
+                         return dm5System != null
+                                 && !dm5System.getStatus().isEnabled()
+                                 && (!dm26System.getStatus().isComplete() || dm26System.getStatus().isEnabled());
```

Comment:
The english reads as !(dm26.isComplete && !dm26.isEnabled), then that is equivalent to (!dm26.isComplete || dm26isEnabled), right?

Reply:
Yes. I updated the logic to more closely match the requirements.